### PR TITLE
update few deps

### PR DESCRIPTION
**Pull request by klensy**
Updates few crates:

* openssl-src v111.18.0+1.1.1n -> v111.22.0+1.1.1q: fixes few CVE's (https://www.openssl.org/news/vulnerabilities-1.1.1.html: https://cve.org/CVERecord?id=CVE-2022-1292 https://cve.org/CVERecord?id=CVE-2022-2068 https://cve.org/CVERecord?id=CVE-2022-2097)

* openssl-probe v0.1.2 -> v0.1.5 updates ancient (2017) crate (https://github.com/alexcrichton/openssl-probe/compare/0.1.2...0.1.5). Adds support to search cert for additional platforms.

* indoc v1.0.3 -> v1.0.6 (https://github.com/dtolnay/indoc/compare/1.0.3...1.0.6) Nothing special changed, removes unindent v0.1.7
 
* bstr v0.2.13 -> v0.2.17 (https://github.com/BurntSushi/bstr/compare/0.2.13...0.2.17) Few speedups (https://github.com/BurntSushi/bstr/commit/8e659921312830b91d2a48aafa36fb1a49cba5bc, https://github.com/BurntSushi/bstr/commit/5fcef919adf1b84a960c87126decb0f3ef8c2e61) and bugfix (https://github.com/BurntSushi/bstr/commit/b2111b6bbf2c9a819fb1338aa81bd099874106a1)

* crc32fast v1.2.0 -> v1.3.2 (https://github.com/srijs/rust-crc32fast/compare/v1.2.0...v1.3.2) Speedup debug (https://github.com/srijs/rust-crc32fast/commit/e61ce6a39bbe9da495198a4037292ec299e8970f)

* diff v0.1.12 -> v0.1.13 (https://github.com/utkarshkukreti/diff.rs/compare/0.1.12...0.1.13) Few optimizations (https://github.com/utkarshkukreti/diff.rs/pull/16, https://github.com/utkarshkukreti/diff.rs/commit/0f0aa580f197af7dab9ab905568f70a2387e23bc)
 
* ignore v0.4.17 -> v0.4.18 (it's hard to get usable diff, but most notable perf change is https://github.com/BurntSushi/ripgrep/commit/a28e664abd0d8912e4a2d85039fe282b9d37b994)

* globset v0.4.5 -> v0.4.9

* regex v1.5.5 -> v1.5.6 few bugfixes (https://github.com/rust-lang/regex/blob/1.5.6/CHANGELOG.md#156-2022-05-20). There exist 1.6.0 version, but it's too fresh.

### CODE REVIEW

**Comment by Mark-Simulacrum** [comment]
You'll probably need to adjust the src/tools/rustc-workspace-hack/Cargo.toml to enable that feature for memchr across our workspace.

**Comment by klensy** [comment]
Looks like #99771 recently bumped memchr (https://github.com/BurntSushi/memchr/compare/2.4.1...2.5.0) for stdlib too, so maybe that interfere with this PR crates' version/feature set.

File: src/tools/rustc-workspace-hack/Cargo.toml [hunk 1]
```diff
@@ -71,16 +71,20 @@ features = [
...
 curl-sys = { version = "0.4.13", features = ["http2", "libnghttp2-sys"], optional = true }
 crossbeam-utils = { version = "0.8.0", features = ["nightly"] }
 libc = { version = "0.2.79", features = ["align"] }
 # Ensure default features of libz-sys, which are disabled in some scenarios.
 libz-sys = { version = "1.1.2" }
+
+# looks like the only user of deprecated `use_std` feature is `combine`, so this
+# can be removed if/when https://github.com/Marwes/combine/pull/348 be merged and released.
+memchr = { version = "2.5", features = ["std", "use_std"] }
 # same for regex
```
Comment (mqudsi) [inline src/tools/rustc-workspace-hack/Cargo.toml:86]: Nitpick: This comment is now left with an invalid reference! It previously referred to "the previous comment" which spoke about the default features of libz-sys, but you added a comment in between the two and now it looks like "same for regex" means "use_std is deprecated ..."

Comment (klensy) [reply src/tools/rustc-workspace-hack/Cargo.toml:86]: Looks like it.

Comment (RalfJung) [reply src/tools/rustc-workspace-hack/Cargo.toml:86]: So, could someone please file a follow-up PR to fix that comment?

Comment (mqudsi) [reply src/tools/rustc-workspace-hack/Cargo.toml:86]: Sure. I assigned #100070 to you.

**Pull request merged**